tests/unit/fem/test_pa_kernels.cpp

```diff
@@ -271,14 +266,14 @@ TEST_CASE("Nonlinear Convection", "[PartialAssembly], [NonlinearPA]")
 template <typename INTEGRATOR>
 double test_vector_pa_integrator(int dim)
 {
-   Mesh *mesh =
+   Mesh mesh =
       (dim == 2) ?
-      new Mesh(2, 2, Element::QUADRILATERAL, 0, 1.0, 1.0):
-      new Mesh(2, 2, 2, Element::HEXAHEDRON, 0, 1.0, 1.0, 1.0);
+      Mesh::MakeCartesian2D(2, 2, Element::QUADRILATERAL, 0, 1.0, 1.0):
+      Mesh::MakeCartesian3D(2, 2, 2, Element::HEXAHEDRON, 1.0, 1.0, 1.0);
```

Comment:
I'm not sure copy-elision will be used here.

It's just a test but if we think of tests as simple examples that users may copy, it may not be a good example to copy.

Reply:
I don't think we need to worry about copy elision, since the right-hand side is movable, the compiler will (at worst) use the move constructor.

On the other hand, in these cases, I believe the compiler is actually smart enough to do *move elision*. That may not be guaranteed by the standard, but the move constructor is so cheap that it doesn't really matter.

As mentioned above, the copy assignment operator is implicitly deleted, so it can't be called. The copy constructor is marked explicit, so it won't be called implicitly when using an assignment expression.

This means that if, because of some mistake, we accidentally did something that should trigger a copy instead of (at worst) a move, the compiler will give an error.